controllers/awsmachine_controller_test.go

```diff
@@ -1004,6 +1021,367 @@ func TestAWSMachineReconciler(t *testing.T) {
+				AvailabilityZone: "us-east-1a",
+			}},
+		}}})
+		g.Expect(err).To(BeNil())
+		cs.Cluster = &clusterv1.Cluster{ObjectMeta: metav1.ObjectMeta{Name: "test-cluster"}}
+		cs.AWSCluster.Status.Network.APIServerELB.DNSName = DNSName
+		cs.AWSCluster.Status.Network.SecurityGroups = map[infrav1.SecurityGroupRole]infrav1.SecurityGroup{
+			infrav1.SecurityGroupNode: {
+				ID: "1",
+			},
+			infrav1.SecurityGroupLB: {
+				ID: "2",
+			},
+			infrav1.SecurityGroupControlPlane: {
+				ID: "3",
+			}}
+		ms, err := getMachineScope(cs, awsMachine)
+		g.Expect(err).To(BeNil())
+
+		ms.Machine.Spec.Bootstrap.DataSecretName = aws.String("bootstrap-data")
+		ms.Machine.Spec.Version = aws.String("test")
+		ms.AWSMachine.Spec.Subnet = &infrav1.AWSResourceReference{ID: aws.String("subnet-1")}
+		ms.AWSMachine.Status.InstanceState = &infrav1.InstanceStateRunning
+		ms.Machine.Labels = map[string]string{clusterv1.MachineControlPlaneLabelName: ""}
+
+		ec2Svc := ec2Service.NewService(cs)
+		ec2Svc.EC2Client = ec2Mock
+		reconciler.ec2ServiceFactory = func(scope scope.EC2Scope) services.EC2MachineInterface {
+			return ec2Svc
+		}
+
+		elbSvc := elbService.NewService(cs)
+		elbSvc.EC2Client = ec2Mock
+		elbSvc.ELBClient = elbMock
+		reconciler.elbServiceFactory = func(scope scope.ELBScope) services.ELBInterface {
+			return elbSvc
+		}
+
+		reconciler.secretsManagerServiceFactory = func(clusterScope cloud.ClusterScoper) services.SecretInterface {
+			return secretMock
+		}
+
+		_, err = reconciler.reconcileNormal(ctx, ms, cs, cs, cs)
+		g.Expect(err).Should(HaveOccurred())
+		expectConditions(g, ms.AWSMachine, []conditionAssertion{{infrav1.InstanceReadyCondition, corev1.ConditionTrue, "", ""}})
+		g.Expect(ms.AWSMachine.Finalizers).Should(ContainElement(infrav1.MachineFinalizer))
+	})
+	t.Run("Should fail in reconciling control-plane machine deletion while terminating instance ", func(t *testing.T) {
+		g := NewWithT(t)
+		mockCtrl = gomock.NewController(t)
+		ec2Mock := mock_ec2iface.NewMockEC2API(mockCtrl)
+		elbMock := mock_elbiface.NewMockELBAPI(mockCtrl)
+
+		expect := func(m *mock_ec2iface.MockEC2APIMockRecorder, e *mock_elbiface.MockELBAPIMockRecorder) {
+			mockedDescribeInstanceCalls(m)
+			mockedDeleteLBCalls(e)
+			m.TerminateInstances(
+				gomock.Eq(&ec2.TerminateInstancesInput{
+					InstanceIds: aws.StringSlice([]string{"id-1"}),
+				}),
+			).
+				Return(nil, errors.New("Failed to delete instance"))
+		}
+		expect(ec2Mock.EXPECT(), elbMock.EXPECT())
+
+		ns, err := testEnv.CreateNamespace(ctx, fmt.Sprintf("integ-test-%s", util.RandomString(5)))
+		g.Expect(err).To(BeNil())
+
+		setup(t, g)
+		awsMachine := getAWSMachine()
+		awsMachine.Namespace = ns.Name
+		createAWSMachine(g, awsMachine)
+
+		defer teardown(g)
+		defer t.Cleanup(func() {
+			g.Expect(testEnv.Cleanup(ctx, awsMachine, ns)).To(Succeed())
+		})
+
+		cs, err := getClusterScope(infrav1.AWSCluster{ObjectMeta: metav1.ObjectMeta{Name: "test"}})
+		g.Expect(err).To(BeNil())
```

Comment:
Same comment as given earlier on `cs and ms` objects for checking the `!=nil`

Reply:
not applicable for ms